internal/services/keyvault/client/helpers.go

```diff
@@ -113,7 +113,7 @@ func (c *Client) KeyVaultIDFromBaseUrl(ctx context.Context, resourcesClient *res
 	}
 
 	filter := fmt.Sprintf("resourceType eq 'Microsoft.KeyVault/vaults' and name eq '%s'", *keyVaultName)
-	result, err := resourcesClient.ResourcesClient.List(ctx, filter, "", utils.Int32(5))
+	result, err := resourcesClient.ResourcesClient.List(ctx, filter, "", nil)
```

Comment:
@sinbai this query above is filtering for the resource type and exact name, with a maximum of 5 results - since this is an exact match there should only ever be a single result (or no results) returned - removing the value for the `top` parameter will instead return (I believe) 50-100 results by default.

As such I'm a little confused as to how this change will change things, given we're expecting a single result here?

Reply:
In fact, the behavior of the Resource API is different with or without the top parameter. The resource API that uses top+filter first top x from all the resources and then filter them in the top x resources, it will return {"value":[],"nextLink":"xxxxx"} if it does not exist in the first top x. The number of request depends on quantity of resources and the location of the retrieved resource. if there is no top parameter, it will return the value(or no results) directly, and only one request is required. Try the following link may be the proof of above:
https://docs.microsoft.com/en-us/rest/api/resources/resources/list#code-try-0

Reply:
Totally agree with a @sinbai

query with a $top returns {{"value": []}, "nextLink": "url"}
query without $top returns {"value": "url" }, 

I will open a ticket to Azure, but I want to know why it's used.
We put a strict name of Azure Key Vault in the query
There is no chance to create multiple resources with the same name under the same subscription

Reply:
@tombuildsstuff  do you have any updates around the status of the pull request?